script/array.js

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable */
+function CreateGuid() {
+  function p8(s) {
+    const p = (`${Math.random().toString(10)}000000000`).substr(2, 8);
+    return s ? +p.substr(0, 4) + p.substr(4, 4) : p;
+  }
+  return p8() + p8(true);
+}
+
+class BooksArray {
+  #books = []; // private encapsulation
+
+  AddBook(addTitle, addAuthor) {
+    const guindex = CreateGuid();
+    const book = {
+      index: guindex,
+      title: addTitle,
+      author: addAuthor,
+    };
+    this.#books.push(book);
+    return book;
+  }
+
```

Comment:
- [ ] I am not able to add a book in the books array please check your `addBook` method and fix where the issue is
![arre](https://user-images.githubusercontent.com/44745855/133495573-f54631bb-67a9-4f7e-ad96-571b2d410256.png)

This is the error am getting when I try to add a book

Reply:
- [ ] It means if I can't add a book I can't test `remove` method please fix your methods

Reply:
was certain uncompatibility in some browsers, it was fixed